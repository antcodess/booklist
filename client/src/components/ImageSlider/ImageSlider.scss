.slider {
  margin-bottom: 60px;

  .image-placeholder {
    position: relative;

    .custom-button {
      position: absolute;
      width: 35%;
      height: 15%;
      background-color: white;
      color: black;
      font-size: 0.94rem;
      // padding: 0px 80px;
      left: 33%;
      bottom: 4%;
      z-index: 5;
      cursor: pointer;
      white-space: nowrap;

      // &.slider-one {
      //   left: 31%;
      // }

      &:hover {
        color: white;
        background-color: black;
      }
    }
    .slider-image {
      border-radius: 10px;
      max-width: 100%;
    }
  }

  .slick-dots li button:before {
    font-size: 13px;
    line-height: 40px;
  }

}
// .slick-prev:before, .slick-next:before {
//   color: black;
// }

@media only screen and (min-width: 1000px) {
  .slider {
    .image-placeholder {
      .custom-button {
        font-size: 1.0rem;
      }
    }
  }
}

@media only screen and (min-width: 1100px) {
  .slider {
    .image-placeholder {
      .custom-button {
        font-size: 1.1rem;
      }
    }
  }
}

@media only screen and (min-width: 1200px) {
  .slider {
    .image-placeholder {
      .custom-button {
        font-size: 1.2rem;
      }
    }
  }
}

// @media only screen and (min-width: 1500px) {
//   .slider {
//     .image-placeholder {
//       .custom-button {
//         font-size: 1.6rem;
//       }
//     }
//   }
// }

@media only screen and (max-width: 830px) {

  .slider {
    .image-placeholder {
      .custom-button {
        font-size: 0.84rem;
      }
    }
  }
}

@media only screen and (max-width: 600px) {
  .slider {
    .image-placeholder {
      .custom-button {
        font-size: .64rem;
        padding: 0px 10px;
      }
    }
  }
}

@media only screen and (max-width: 500px) {
  .slider {
    .image-placeholder {
      .custom-button {
        display: none;
      }
    }
  }
}

