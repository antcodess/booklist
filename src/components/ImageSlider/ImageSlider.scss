.slider {
  margin-bottom: 60px;

  .image-placeholder {

    .slider-image {
      border-radius: 10px;
      max-width: 100%;
    }
  }

  .slick-dots li button:before {
    font-size: 13px;
    line-height: 40px;
  }

}
// .slick-prev:before, .slick-next:before {
//   color: black;
// }

