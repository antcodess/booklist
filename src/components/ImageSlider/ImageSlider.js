import React from 'react';
import Slider from 'react-slick';
import 'slick-carousel/slick/slick.css';
import 'slick-carousel/slick/slick-theme.css';

import photoOne from '../../assets/photo1.jpg';
import photoTwo from '../../assets/photo2.jpg';
import photoThree from '../../assets/photo3.jpg';
import './ImageSlider.scss';

const ImageSlider = () => {
  
  const settings = {
    dots: true,
    infinite: true,
    speed: 500,
    slidesToShow: 1,
    slidesToScroll: 1,
    autoplay: true,
    autoplaySpeed: 5000,
    pauseOnHover: true
  };

  return (
    <section className='slider'>
      <Slider {...settings } >
        <div className='image-placeholder'>
          <img src={photoOne} alt='Books' />
        </div>
        <div className='image-placeholder'>
          <img src={photoTwo} alt='Books' />
        </div>
        <div className='image-placeholder'>
          <img src={photoThree} alt='Books' />
        </div>
      </Slider>
    </section>
  );
};

export default ImageSlider;