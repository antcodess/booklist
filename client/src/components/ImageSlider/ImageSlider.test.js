import React from 'react';
import { shallow } from 'enzyme';
import { ImageSlider } from './ImageSlider';

describe('ImageSlider Component', () => {
  let wrapper;
  let mockHistory;
  let dummyFunction = jest.fn();
  
  beforeEach(() => {
    
    mockHistory = { push: jest.fn() };

    const mockProps = { history: mockHistory };

    wrapper = shallow(<ImageSlider { ...mockProps } />);
  });

  it('should render ImageSlider Component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call history.push when clicked for image-slider-placeholder1 div', () => {
    wrapper.find('#image-slider-placeholder1').simulate('click');
    expect(mockHistory.push).toHaveBeenCalled();
  });

  it('should call history.push when clicked for image-slider-placeholder2 div', () => {
    wrapper.find('#image-slider-placeholder2').simulate('click');
    expect(mockHistory.push).toHaveBeenCalled();
  });

  it('should call history.push when clicked for image-slider-placeholder3 div', () => {
    wrapper.find('#image-slider-placeholder3').simulate('click');
    expect(mockHistory.push).toHaveBeenCalled();
  });
});