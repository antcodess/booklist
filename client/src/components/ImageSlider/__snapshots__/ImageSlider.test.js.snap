// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageSlider Component should render ImageSlider Component 1`] = `ShallowWrapper {}`;
